### PR TITLE
Hotfix/fix api not calling problem

### PR DESCRIPTION
**Pull request by Pjaijai**
I found the problem is in auth provider:

```

useEffect(() => {
    const {
      data: { subscription: authListener },
    } = supabase.auth.onAuthStateChange(async (event, session) => {
      if (session && session.access_token !== accessToken) {
        router.refresh()
      } else if (
        session &&
        session.access_token === accessToken &&
        user.isSignIn !== true
      ) {
        const { data, error }: any = await supabase
          .from("user")
          .select(
            "email,uuid,
            username,
            avatar_url
            "
          )
          .eq("uuid", session.user.id)
          .single()

        setUserState({
          email: data.email,
          uuid: data.uuid,
          username: data.username,
          photoUrl: data.avatar_url,
        })
      } else {
        reSetUserState()
      }
    })

    return () => {
      authListener?.unsubscribe()
    }
  }, [accessToken, supabase, router])

```

It seems like the onAuthStateChange cannot work properly with async call back ? The type of it :

```
 onAuthStateChange(
    callback: (event: AuthChangeEvent, session: Session | null) => void | Promise<void>
  ): {
    data: { subscription: Subscription }
  } {
    const id: string = uuid()
    const subscription: Subscription = {
      id,
      callback,
      unsubscribe: () => {
        this._debug('#unsubscribe()', 'state change callback with id removed', id)

        this.stateChangeEmitters.delete(id)
      },
    }

    this._debug('#onAuthStateChange()', 'registered callback with id', id)

    this.stateChangeEmitters.set(id, subscription)
    ;(async () => {
      await this.initializePromise

      await this._acquireLock(-1, async () => {
        this._emitInitialSession(id)
      })
    })()
```

  So i make a hot fix for it. Extracting the asyc api out from onAuthStateChange. It seems working fine??

### CODE REVIEW

File: client/components/providers/auth.tsx [hunk 1]
```diff
@@ -59,6 +45,44 @@ const AuthProvider: FunctionComponent<IAuthProviderProps> = ({
...
+              description: "好似有啲錯誤，如果試多幾次都係咁，請聯絡我🙏🏻",
+              variant: "destructive",
+            })
+          }
+        } catch (error) {
+          return toast({
+            title: "登入出事！",
+            description: "好似有啲錯誤，如果試多幾次都係咁，請聯絡我🙏🏻",
+            variant: "destructive",
+          })
```
Comment (yaucp) [inline client/components/providers/auth.tsx:76]: Vercel 想問有冇logger 如果有嘅話可以log error

Comment (Pjaijai) [reply client/components/providers/auth.tsx:77]: 你意思係throw error ? 好似冇throw 到，update 咗

Comment (Pjaijai) [reply client/components/providers/auth.tsx:77]: 冇log，個api冇行到

Comment (yaucp) [reply client/components/providers/auth.tsx:77]: i mean if we are using any runtime logger at all, we should include the error and log them for investigation? if we dun have runtime logger jau msw

Comment (Pjaijai) [reply client/components/providers/auth.tsx:77]: Vercel and Supabase do have the thing. But I am considering adding something like Sentry later.

Comment (yaucp) [reply client/components/providers/auth.tsx:77]: Okok

**Review by yaucp — APPROVED** [review]

**Pull request merged**